sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuAvroScan.scala

```diff
@@ -769,14 +782,15 @@ class GpuMultiFileCloudAvroPartitionReader(
                   batchRowsNum <= maxReadBatchSizeRows)
 
               // One batch is done
-              hostBuffers += ((hmb, out.getPos))
+              if (optOut.nonEmpty) {
```

Comment:
Lines 785 to 787 can be simplified as 
```suggestion
          optOut.foreach(out => hostBuffers += ((optHmb.get, out.getPos)))    
```

Reply:
done